cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -78,10 +78,12 @@ type Networking struct {
 }
 
 type Etcd struct {
-	Endpoints []string
-	CAFile    string
-	CertFile  string
-	KeyFile   string
+	Endpoints    []string
+	CAFile       string
+	CertFile     string
+	KeyFile      string
+	ListenURL    string
```

Comment:
I think we really should go with a generic Args here instead of two hard-coded string fields.
Imagine I want to set the token authz mechanism. Eventually we would duplicate all etcd options, instead we should go with a more dynamic route with `map[string]string` as the other control plane components